src/store/create-conversation/saga.ts

```diff
@@ -102,6 +103,13 @@ const PREVIOUS_STAGES = {
   [Stage.GroupDetails]: Stage.StartGroupChat,
 };
 
+function* handleCreateConversation() {
```

Comment:
Added this `handleCreateConversation` function to prevent changing to much around the setting of the `Stages`. Previously, when `this.props.stage === SagaStage.None` the action `createConversation/create` would not dispatch and therefore the direct message chat would not pop up in the UI. Not sure if this is the best approach, but appears to work well with little change to support code.

Reply:
I think we need to tweak this a little. My concern is that we'll get into a state where 2 things are listening for this event at the same time. I _think_ one will eat it and the other will be left in an inconsistent state.

Might be worth pairing for a bit here but, at a quick glance, what you probably want is that in the "normal" case we're listening for either a createconversation event OR a `SagaActionTypes.Start` event. It'd be a `race` call because you want whichever comes first. If you get the createconvo event then do the create convo...if you get a start event then drop into the startConversation flow.

Reply:
I looked into this a little and followed your suggestion to use a `race` call. I updated the `saga` function to listen for either a `SagaActionTypes.Start` or `SagaActionTypes.CreateConversation` event, so whichever event comes first, race should cancel the other. If `SagaActionTypes.Start` comes first, the saga will call `startConversation`; if `SagaActionTypes.CreateConversation` comes first, it will call `createConversation` with the event payload, as suggested! Let me know what you think 💪  Thanks for the guidance above!